### ISSUE TITLE
converting  galactic cartesian SkyCoord to spherical and also to ecliptical

### ISSUE THREAD

**Issue by richardgmcmahon** [astropy/astropy#7377]
![plot_galactic_radec](https://user-images.githubusercontent.com/2388403/38767278-287409fa-3fd6-11e8-87a1-af7f5d1bbe5f.png)
Plotting random data in Aitoff projection: plotting galactic coordindates

I am strugglig with converting a SkyCoord in galactic cartesian to galactic spherical

I want to create a set of  example plots in equatotal, galactic and ecliptic cordinates that are 
not just a random set of uniform points. I have started from the this example:

Example 1: Plotting random data in Aitoff projection

http://docs.astropy.org/en/stable/coordinates/skycoord.html#example-2-plotting-star-positions-in-bulge-and-disk

This creates a SkyCoord using u, v, w 

coords_disk = SkyCoord(disk,
                     representation='cartesian',
                     frame='galactic')

I have stuggled with getting galactic cords in long, lat or l, b from this. The u, v, w, cartesian form is being used to that a simple disk and bulge distribution can be created using np.random.multivariate_normal

print(coords_disk)
<SkyCoord (Galactic): (u, v, w) in rad

print(coords_disk.u) works

print(cords_disk.l) or .lon gives an error as expected

So I tried changing the representation:

coords_disk.representation = 'spherical'
print(coords_disk)
<SkyCoord (Galactic): (l, b, distance) in (deg, deg, rad)
print(coords_disk.l) work.

Is there a way to convert the SkyCoord from cartesian to sperical since that would be more 
convenient since I want to also plot ecliptic cordinates.

I have been able to this:

coords_disk.representation = 'spherical'
coords_bulge.representation = 'spherical'

l_disk_rad = coords_disk.l.wrap_at(180 * u.deg).radian
b_disk_rad = coords_disk.b.radian

l_bulge_rad = coords_bulge.galactic.l.wrap_at(180 * u.deg).radian
b_bulge_rad = coords_bulge.galactic.b.radian

![plot_galactic_galactic](https://user-images.githubusercontent.com/2388403/38767337-2dc90bb6-3fd7-11e8-8cac-225ce4274c86.png)

How do I get ecliptic?

OK, maybe I have answered my own question:

from astropy.coordinates import BarycentricTrueEcliptic
coords_disk_ecliptic = coords_disk.transform_to(BarycentricTrueEcliptic)

l_disk_rad = coords_disk_ecliptic.lon.wrap_at(180 * u.deg).radian
b_disk_rad = coords_disk_ecliptic.lat.radian

**Comment by adrn** [comment]
That all looks correct to me!

**Comment by astrofrog** [comment]
@richardgmcmahon - is there anything in the documentation that could have been clearer?

**Comment by richardgmcmahon** [comment]
OK, it looks like I solved my own problem.  I can see the power and flexibility of skycoord and appreciate the thoughts behind it.  

The interplay between frame and representation caused me some confusion and I have many means of experience from the preparation of observations.

Maybe this example could be extended to include transforming to galactic longitude and latitude as in my scenario since would cover the relevance and difference between 'frame' and 'representation'.

http://docs.astropy.org/en/stable/coordinates/skycoord.html#example-2-plotting-star-positions-in-bulge-and-disk

I have pushed my example to:

https://github.com/richardgmcmahon/astropy_examples
see: plot_galactic*
